clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -368,4 +368,34 @@ def CIR_VisibilityAttr : CIR_Attr<"Visibility", "visibility"> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// ExtraFuncAttributesAttr
+//===----------------------------------------------------------------------===//
+
+def ExtraFuncAttributesAttr : CIR_Attr<"ExtraFuncAttributes", "extra"> {
```

Comment:
@xlauko and I start discussing about whether we really need `ExtraFuncAttributesAttr`. Maybe this is a good point in time to decide on it, and perhaps we don't need to introduce this upstream. The original idea was to use this to share attributes that later in LLVM could be placed in both functions and calls, but currently there isn't anything special about it (IIRC) and LLVMDialect uses a more natural approach to represent these, seems like we could do similar?

Reply:
+1 for not upstreaming this if it isn't needed.

Reply:
LGTM to not upstreaming this if it's unnecessary, but then should we still upstreaming the `#cir.nothrow` attribute? Now `#cir.nothrow` is defined as a `CIRUnitAttr`, but maybe we could just use a simple `UnitAttr` for it instead?

Reply:
Good catch, I think `CIRUnitAttr` is in a similar boat!

Reply:
Yeah, I am trying to get it work without custom unit attributes and `ExtraFuncAttributesAttr` but it will take me a bit more time. I first need to finish assembly format for FuncOp, so its easier to pull out these extra attrs. But I am almost there.